lib/util/host-rules.ts

```diff
@@ -8,47 +7,54 @@ import { parseUrl, validateUrl } from './url';
 
 let hostRules: HostRule[] = [];
 
-const matchFields = ['matchHost', 'hostName', 'domainName', 'baseUrl'];
+const legacyHostFields = ['hostName', 'domainName', 'baseUrl'];
 
 export function add(params: HostRule): void {
-  const matchedFields = matchFields.filter((field) => params[field]);
-  if (matchedFields.length > 1) {
-    throw new Error(
-      `hostRules cannot contain more than one host-matching field. Found: [${matchedFields.join(
-        ', '
-      )}]`
+  const rule = clone(params);
+  const matchedFields = legacyHostFields.filter((field) => rule[field]);
```

Comment:
I've kept this migration code here just in case there are any "missed" migrations anywhere. The implications would be too high if hostRules stopped working.